### PR TITLE
Remove LoDash dependency

### PR DESCRIPTION
**Pull request by ChrisMBarr**
My Angular project uses ChordsheetJS and it gave this this warning upon build about Chordsheet using LoDash, which is not an ESM module

2024-11-29 11_32_59-package json - LyricConverter - Visual Studio Code

Looking into it, I saw ChordsheetJS only used LoDash's `_.get()` method, and it only ever used it to access a single property.  Seems like an easy one to just access directly with a helper method that has a default/fallback of an empty string

### CODE REVIEW

**Comment by ChrisMBarr** [comment]
Not sure what the deal is with that failure, it seems unrelated to any of the changes I made since all the tests pass

**Comment by martijnversluis** [comment]
Hey @ChrisMBarr. Thanks so much for taking the time to look at this! I do agree it would be great to drop all runtime dependencies. 

Unfortunately, being able to get any key from metadata is critical to support the chordpro standard. For example, there are official [metadata directives](https://www.chordpro.org/chordpro/chordpro-directives/#meta-data-directives) and  you can define and use arbitrary values (like `{x_myapp_some_setting}`). It is also possible to get array values. See also the docs for [metadata.get](https://github.com/martijnversluis/ChordSheetJS#get). 

This means, if we remove the dependency, that we have to cover the existing functionality.

If you would like help with that, surely let me know! 🙏

**Comment by ChrisMBarr** [comment]
Oh ok, I didn't realize that `get` was a public method!  Ok, I have something I can fall back to using something like this: https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_get

**Comment by martijnversluis** [comment]
@ChrisMBarr Would you like to implement that? Please let me know might you need help 👍

**Comment by ChrisMBarr** [comment]
Ok, this should be good to go now.  Unit tests pass, however I did notice that there are no unit tests to verify how `.get()` itself works, to be able to access arbitrary properties or access array items.  It should work, but it might be a good idea to add those just in case.

**Comment by martijnversluis** [comment]
Thanks for your work! 🙏 

Yes, apparently unit tests are missing, I would've expected those would fail when removing the lodash package.

Would you like me to add those tests?

**Comment by ChrisMBarr** [comment]
I started to look into this... but from what I can tell [the `metadata.get` method as documented here](https://martijnversluis.github.io/ChordSheetJS/classes/Metadata.html#get) uses [a completely different method](https://github.com/martijnversluis/ChordSheetJS/blob/master/src/chord_sheet/metadata.ts#L108-L118) than the one I re-wrote, and that one [already has tests written for it](https://github.com/martijnversluis/ChordSheetJS/blob/master/test/chord_sheet/metadata.test.ts#L39)

Is this one that previously used LoDash even public at all?  Does it even need any tests?  It appear that the only thing that uses this are the internal calls to `configuration.get('metadata.separator')` which could just be re-written as `configuration.metadata.separator` to directly access what is needed instead of doing a weird string path thing.

**Comment by martijnversluis** [comment]
> I started to look into this... but from what I can tell [the `metadata.get` method as documented here](https://martijnversluis.github.io/ChordSheetJS/classes/Metadata.html#get) uses [a completely different method](https://github.com/martijnversluis/ChordSheetJS/blob/master/src/chord_sheet/metadata.ts?rgh-link-date=2024-12-08T01%3A24%3A21Z#L108-L118) than the one I re-wrote, and that one [already has tests written for it](https://github.com/martijnversluis/ChordSheetJS/blob/master/test/chord_sheet/metadata.test.ts?rgh-link-date=2024-12-08T01%3A24%3A21Z#L39)

You're completely right. I really thought Lodash was used in metadata, not in configuration 😅 

> Is this one that previously used LoDash even public at all? Does it even need any tests? It appear that the only thing that uses this are the internal calls to `configuration.get('metadata.separator')` which could just be re-written as `configuration.metadata.separator` to directly access what is needed instead of doing a weird string path thing.

It think you're right that the usages can be rewritten. The class could use some tests, but that was also the case with Lodash. I'm fine with `configuration.metadata.separator` 👍

**Comment by ChrisMBarr** [comment]
Alright, that's now in so there is no more `configuration.get` method at all! There's just a getter method/property that will always return a string. Much simpler now, and one less dependency

File: src/formatter/configuration/configuration.ts [hunk 1]
```diff
@@ -53,6 +51,10 @@ class Configuration {
 
   delegates: Partial<Record<ContentType, Delegate>>;
 
+  get separator(): string {
```
Comment (martijnversluis) [inline src/formatter/configuration/configuration.ts:54]: I would prefer `metadataSeparator` as this is used to join multiple metadata values.

File: package.json [hunk 2]
```diff
@@ -74,8 +73,5 @@
     "test": "yarn unibuild lint && yarn unibuild test",
     "unibuild": "tsx ./unibuild.ts"
   },
-  "dependencies": {
```
Comment (martijnversluis) [inline package.json:77]: Whoohoo, no more dependencies! 🙌

**Review by martijnversluis — CHANGES_REQUESTED** [review]
Great job! I would suggest one rename, other than that this is good to go! 👍

**Review by martijnversluis — APPROVED** [review]

**Comment by martijnversluis** [comment]
Awesome @ChrisMBarr! Thanks for contributing 🎉

**Pull request merged**